synapse/config/workers.py

```diff
@@ -215,12 +216,55 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         )
 
         # A map from instance name to host/port of their HTTP replication endpoint.
+        # Check if the main process is declared. Inject it into the map if it's not,
+        # based first on if a 'main' block is declared then on 'worker_replication_*'
+        # data. If both are available, default to instance_map. The main process
+        # itself doesn't need this data as it would never have to talk to itself.
+        instance_map: Dict[str, Any] = config.get("instance_map", {})
+
+        if instance_map and self.instance_name is not MAIN_PROCESS_INSTANCE_NAME:
+            # The host used to connect to the main synapse
+            main_host = config.get("worker_replication_host", None)
+
+            # The port on the main synapse for HTTP replication endpoint
+            main_port = config.get("worker_replication_http_port")
+
+            # The tls mode on the main synapse for HTTP replication endpoint.
+            # For backward compatibility this defaults to False.
+            main_tls = config.get("worker_replication_http_tls", False)
+
+            # For now, accept 'main' in the instance_map, but the replication system
+            # expects 'master', force that into being until it's changed later.
+            if MAIN_PROCESS_INSTANCE_MAP_NAME in instance_map:
+                instance_map.setdefault(MAIN_PROCESS_INSTANCE_NAME, {})
+                for k, v in instance_map[MAIN_PROCESS_INSTANCE_MAP_NAME].items():
+                    instance_map[MAIN_PROCESS_INSTANCE_NAME].setdefault(k, v)
+                del instance_map[MAIN_PROCESS_INSTANCE_MAP_NAME]
```

Comment:
wondering what's wrong with

```suggestion
                instance_map[MAIN_PROCESS_INSTANCE_NAME] = instance_map[MAIN_PROCESS_INSTANCE_MAP_NAME]
                del instance_map[MAIN_PROCESS_INSTANCE_MAP_NAME]
```

Reply:
It's been a few weeks, but IIRC it was giving me a `KeyError` when I tried that(I can double check, it may have been another reason). Once the migration from `master` to `main` is complete code base wide, this will all be deleted anyways. The only reason it exists here at all is because the `ReplicationEndpoint` class still needs it be `master`

Reply:
Whatever I had done before must not have been this. You got it(plus formatting changes cuz lint) in  [f6c24e4](https://github.com/matrix-org/synapse/pull/15491/commits/f6c24e49751b5f9790b0686b6a8d36c27745f8eb)